core/amber/src/main/scala/edu/uci/ics/texera/web/service/WorkflowJobService.scala

```diff
@@ -65,9 +67,12 @@ class WorkflowJobService(
       )
     }
     resultService.attachToJob(workflowInfo, client)
-    val eid = ExecutionsMetadataPersistService.insertNewExecution(workflowContext.wId)
+    if (WorkflowService.userSystemEnabled) {
+      workflowContext.executionID =
+        ExecutionsMetadataPersistService.insertNewExecution(workflowContext.wId)
+    }
     stateStore.jobStateStore.updateState(jobInfo =>
-      jobInfo.withState(READY).withEid(eid).withError(null)
+      jobInfo.withState(READY).withEid(workflowContext.executionID).withError(null)
```

Comment:
if user sys is disabled, wouldn't the `executionID` by default be -1? for this line `withEid(-1)` what is the expected behavior?

Reply:
I feel -1 might be a little bit misleading. for userID we have `Option[UInteger] = None` to indicate no user id. can we use option for `executionID` as well? just a thought.

Reply:
I initially made it optional, but then saw that it is being used in many places just like `wid`. In dev mode `wid` is given a dummy value which is -1 so I did the same for `execID`

Reply:
yes the `wid` was really a dirty comprise already, because it needs to have an id to be able to connect a web socket, while `None` does not work for that.  I'd suggest we use `Option` if possible.

Reply:
nvm, I checked the execution id really needs a dummy value... so let's keep it this way as well for now.